src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -598,7 +599,7 @@ private async Task InitializeAsync(bool ignoreFailures, IEnumerable<Configuratio
             var serverData = new Dictionary<string, ConfigurationSetting>(StringComparer.OrdinalIgnoreCase);
 
             // Use default query if there are no key-values specified for use other than the feature flags
-            bool useDefaultQuery = !_options.KeyValueSelectors.Any(selector => !selector.KeyFilter.StartsWith(FeatureManagementConstants.FeatureFlagMarker));
+            bool useDefaultQuery = !_options.KeyValueSelectors.Any(selector => (!string.IsNullOrEmpty(selector.KeyFilter) && !selector.KeyFilter.StartsWith(FeatureManagementConstants.FeatureFlagMarker)) || !string.IsNullOrEmpty(selector.SnapshotName));
```

Comment:
Can you multi-line this. It's a bit long.

Reply:
It seems like it should be

`bool useDefaultQuery = !_options.KeyValueSelectors.Any(selector => selector.KeyFilter == null || !selector.KeyFilter.StartsWith(FeatureManagementConstants.FeatureFlagMarker));`